### DISCUSSION TITLE
Random extension shows draft pages

### DISCUSSION THREAD

**Discussion by i-do** [datenstrom/community#538] [Report a bug] [answered]
I see the names of draft pages in blog random list and I'm not logged as admin.

Just update:
2020-08-16 14:50:23 info Update extension 'Blog 0.8.10'
2020-08-16 14:50:23 info Update extension 'Core 0.8.20'
2020-08-16 14:50:23 info Update extension 'Draft 0.8.10'
2020-08-16 14:50:23 info Update extension 'Feed 0.8.10'
2020-08-16 14:50:23 info Update extension 'Toc 0.8.5'
2020-08-16 14:50:23 info Update extension 'Update 0.8.33'
2020-08-16 14:50:23 info Update extension 'Wiki 0.8.11'

**Comment by markseuffert** [comment]
If this is a result of latest core update, then this it's probably my fault and I want to help fixing it. Can you describe how to reproduce the problem, can you show the source code to make a blog random list, have any additional information?

**Comment by i-do** [comment]
@markseu, I didn't do nothing special:
1. Use Draft extension 0.8.10
2. Use Blog extension 0.8.10

And the [random /blog/ 10 list] tag.

Made a Blog clean install. Use the "Status: draft" in the post.
Draft extension I didn't touch at all. Just made update from the web panel (via /edit/ mode) as usual.

That's all.

But still see the Draft pages list mixed with the already posted ones.

**Comment by markseuffert** [comment]
Where's the `[random]` shortcut from... [do you use this extension](https://github.com/schulle4u/yellow-extensions-schulle4u/tree/master/random)?

**Comment by i-do** [comment]
Yes, [Mark](@markseu), I use it.

**Comment by i-do** [comment]
Can't understand why I can't see [Draft pages](https://github.com/datenstrom/yellow-extensions/tree/master/source/draft) for Blog.

mysite/edit/drafts/ and mysite/edit/blog/drafts/ doesn't work for me.

**Comment by schulle4u** [comment]
Yes, I can confirm the issue that draft pages show up in random pages selection using the latest core.  Even page content can be displayed. The only thing I did recently was to implement the new API into my extensions, but no functional changes. Could I have missed something? 

The odd thing is that draft and private extensions share similar functions, but private pages are correctly excluded from listings.

**Comment by schulle4u** [comment, marked as answer]
Hi @i-do, please update your site and install the latest random extension. It should now work as expected with draft pages. I never used the draft extension and wasn't aware that my API call was set to display also invisible pages. Seems you are the first with both extensions installed, congratulations for detecting this 5 year old bug. 😀

**Comment by i-do** [comment]
[Steffen](@schulle4u), yes, it's kind of funny. But I used Random for case when I need to see my Draft pages. 
Now Random is working, mysite/edit/blog/drafts/ doesn't. 

Now the big question is: how can I find my drafts?

**Comment by annaesvensson** [comment]
Thanks for your feedback. Please submit a new issue for each problem, [see contributing guidelines](https://datenstrom.se/yellow/help/contributing-guidelines).
